app/models/course_data/courses_users.rb

```diff
@@ -144,6 +144,10 @@ def cleanup
     CourseCleanupManager.new(course, user).cleanup_articles
   end
 
+  def user_page(students)
```

Comment:
I think it makes more sense to implement this as a User instance method that only cares about the single User record and returns the userpage for the corresponding user. Mapping that into an array can be done concisely from WatchlistEdits (eg, ` users.map(&:userpage)`).

Reply:
Okay